storage/innobase/trx/trx0trx.cc

```diff
@@ -2468,3 +2507,63 @@ trx_set_rw_mode(
 		trx->read_view.set_creator_trx_id(trx->id);
 	}
 }
+
+/*************************************************************//**
+Return difference in uncommitted count for a single undo record. */
+ib_int64_t get_diff_from_rec(
+/*============*/
+    trx_undo_rec_t* undo_rec,		/*!< in: undo record */
+    table_id_t table_id)           	/*!< in: table ID */
+{
+	ulint type, cmpl_info;
+	bool updated_extern;
+	undo_no_t undo;
+	table_id_t rec_table_id;
+
+	trx_undo_rec_get_pars(undo_rec, &type, &cmpl_info, &updated_extern, &undo,
+		&rec_table_id);
+
+	if (rec_table_id != table_id)
+		return 0;
+
+	switch (type) {
+		case TRX_UNDO_INSERT_REC:
+			return 1;
+		case TRX_UNDO_UPD_DEL_REC:
+		case TRX_UNDO_DEL_MARK_REC:
+			return -1;
```

Comment:
The `switch` body is indended one level too deep.

I think that for `case TRX_UNDO_UPD_DEL_REC:` we should `return 1`. It is an insert by updating a delete-marked record.